### ISSUE TITLE
Extremely low GPU ultilization

### ISSUE THREAD

**Issue by YangFei1990** [flairNLP/flair#857]
Hello, I was testing the example provided in [TUTORIAL_7_TRAINING_A_MODEL.md](https://github.com/zalandoresearch/flair/blob/v0.4.1/resources/docs/TUTORIAL_7_TRAINING_A_MODEL.md#training-a-sequence-labeling-model) with some small modification. However, I received very low GPU ultilization (around 20%). Also, for the cpu, there are only 1 or 2 cores fully ultilized. 
Here is my code:

```
from flair.data import Corpus 
from flair.data_fetcher import NLPTaskDataFetcher, NLPTask 
from flair.embeddings import TokenEmbeddings, WordEmbeddings, StackedEmbeddings 
from flair.embeddings import CharacterEmbeddings, FlairEmbeddings 
from typing import List

# 1. get the corpus
#corpus: TaggedCorpus = NLPTaskDataFetcher.load_corpus(NLPTask.WNUT).downsample(0.1)
corpus: Corpus = NLPTaskDataFetcher.load_corpus(NLPTask.WNUT_17)
print(corpus)

# 2. what tag do we want to predict?
tag_type = 'ner'

# 3. make the tag dictionary from the corpus
tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
print(tag_dictionary.idx2item)

# 4. initialize embeddings
embedding_types: List[TokenEmbeddings] = [

    WordEmbeddings('glove'),

    # comment in this line to use character embeddings
    CharacterEmbeddings(),

    # comment in these lines to use flair embeddings
    # FlairEmbeddings('news-forward'),
    # FlairEmbeddings('news-backward'),
]

embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

# 5. initialize sequence tagger
from flair.models import SequenceTagger

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        use_crf=True)

# 6. initialize trainer
from flair.trainers import ModelTrainer

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

# 7. start training
trainer.train('resources/taggers/example-ner',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=150)

# 8. plot training curves (optional)
from flair.visual.training_curves import Plotter
plotter = Plotter()
plotter.plot_training_curves('resources/taggers/example-ner/loss.tsv')
plotter.plot_weights('resources/taggers/example-ner/weights.txt')

```

I saw some other related issues like: [issue 37](https://github.com/zalandoresearch/flair/issues/37) and a related pr: [pr 459](https://github.com/zalandoresearch/flair/pull/459) but it has been merged long ago(this Feb). Please let me know if I missed anything. Any help would be much appreciated.

**Comment by Aditya715** [comment]
I've done the same and hit 75% usage of gpu. Almost using the same code. And flair team is working on the gpu utilization but they are in sort of time. Actually when you use it it uses gpu and a single core of cpu, that's why utilization is less. But at least tell me what is the usage of your gpu and your gpu configuration.

**Comment by YangFei1990** [comment]
@Aditya715 I use a single [nvidia tesla v100](https://www.nvidia.com/en-us/data-center/tesla-v100/) GPU with CUDA 10 and my utilization was just around 20%-30%. How did you achieve 75%? Do you have a larger dataset or a different training script?

**Comment by Aditya715** [comment]
No I don't have any larger dataset but yes I've changed the minimum_batch_size=64 and and allow the forward and backward embedding. If you are using smaller dataset then kindly lower the epoch as I've not got the early stopping in flair so far so after a certain epoch your losses will start increasing and learning rate will reduces. Check if your cuda is working in torch or not.

import torch
print(bool(torch.device.is_cuda_available())
if yes then let me know, I will surely run your code on my cuda as soon as it becomes free.

**Comment by YangFei1990** [comment]
@Aditya715 torch.cuda.is_available() is True, please check my code. Thanks!

**Comment by alanakbik** [comment]
A lot of fixes were added over time to increase GPU utilization.
